modules/diskq/diskq-serializer.c

```diff
@@ -45,6 +54,96 @@ _pread_all(int fd, char *buf,  size_t count, off_t offset)
   return retval + _pread_all(fd, buf+retval, count-retval, offset+retval);
 }
 
+static DiskQSerializerCache *
+_cache_new(void)
+{
+  DiskQSerializerCache *self = g_new0(DiskQSerializerCache, 1);
+  self->buffer = g_string_sized_new(CHUNK_SIZE);
+  self->pos = 0;
+
+  return self;
+}
+
+static void
+_cache_free(DiskQSerializerCache *self)
+{
+  g_string_free(self->buffer, TRUE);
+  g_free(self);
+}
+
+static gboolean
+_cache_fetch(DiskQSerializerCache *self, gchar *buf, gsize buflen)
+{
+  if (self->pos + buflen > self->buffer->len)
```

Comment:
Can you move this to a separate function, like `if (!_enough_data_in_cache(self, buflen))`?

Reply:
I split the patchset in two. This will be addressed in the next PR.